examples/common/convert_c4.py

```diff
@@ -4,105 +4,344 @@
 """C4 streaming dataset conversion scripts."""
 import os
 import platform
+import warnings
 from argparse import ArgumentParser, Namespace
-from typing import Dict, Iterable
+from enum import Enum
+from textwrap import dedent
+from typing import Dict, Iterable, Optional
 
 import datasets as hf_datasets
+import numpy as np
 from streaming import MDSWriter
 from torch.utils.data import DataLoader, IterableDataset, get_worker_info
 from tqdm import tqdm
+from transformers import AutoTokenizer, PreTrainedTokenizerBase
+
+
+class ConcatMode(Enum):
+    NO_CONCAT = 'NO_CONCAT'
+    CONCAT_TEXT = 'CONCAT_TEXT'
+    CONCAT_TOKENS = 'CONCAT_TOKENS'
 
 
 def parse_args() -> Namespace:
     """Parse commandline arguments."""
-    args = ArgumentParser()
-    args.add_argument('--out_root', type=str, required=True)
-    args.add_argument('--compression', type=str, default=None)
-    args.add_argument('--splits',
-                      nargs='+',
-                      default=['train', 'train_small', 'val'])
+    parser = ArgumentParser(
+        description=
+        'Convert C4 into MDS format, optionally concatenating and tokenizing')
+    parser.add_argument('--splits',
+                        nargs='+',
+                        default=['train', 'train_small', 'val'])
+    parser.add_argument('--out_root', type=str, required=True)
+    parser.add_argument('--compression', type=str, default=None)
+
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument('--concat_text',
+                       type=int,
+                       help='Number of characters to concatenate to')
+    group.add_argument(
+        '--concat_tokens',
+        type=int,
+        help='Convert text to tokens and concatenate up to this many tokens')
```

Comment:
it looks weird that the formatting is different.
it looks like a pre-commit linter fixed it and we didn't catch it 😅

Reply:
I think this is "intentional" with our formatting penalties. It tries to keep some of the args on the first line and align with them, but that would cause the second help text to go past a threshold?